### PR TITLE
Expose getIdentifierHash in @celo/base

### PR DESCRIPTION
**Pull request by eelanagaraj**
### Description

This PR builds on #10416, which initially tried to remove identifier hashing logic out of `@celo/base` (into `@celo/identity`) and then having `@celo/phone-utils` depend on `@celo/identity`. Unfortunately, that creates a non-trivial, multi-package dependency cycle. More details:
-  dependency cycle (when adding having phone-utils depend on identity SDK, or changing protocol to depend directly on identity SDK): protocol -> `@celo/phone-utils` -> `@celo/identity` -> `@celo/phone-number-privacy-common` -> `@celo/contractkit` (& also `@celo/phone-utils` currently, though this can fairly easily be removed and doesn't cause the main issues here) -> `@celo/phone-utils` & protocol

I moved the minimal amount of things out of `@celo/identity` without making significant changes to the inner functions (i.e. `getPrefixedIdentifier`). The other option if we really want to keep `IdentifierPrefix` in the identity SDK makes things a bit less readable here but should be possible by doing something like:
- base `getIdentifierHash` takes in a `getPrefixedIdentifier` function, and is not as strongly typed (add in `IdentifierPrefix` enforcement only in identity SDK)
- base `getPrefixedIdentifier` is not as strongly typed
- create a `PHONE_NUMBER_PREFIX` constant that is used in `phone-utils` + the `IdentifierPrefix`

^ Downside of this is that if we ever want to use non-PN identifiers in the protocol tests is that we would run into the identity SDK dependency cycle again (i.e. if these identifiers are not accessible through base). For now, I opted keeping these three functions together since I think it's a bit more readable and don't think it makes it too much more difficult for someone to submit a PR adding these back in.

I think another (possibly better) option would be splitting up the identity SDK into two smaller modules, one that is identifier-specific (super lightweight SDK, basically would be the same as the file `packages/sdk/base/src/identifier.ts` but as a separate thing) and then the ODIS-querying stuff. Everything should be able to depend directly on the identifier-specific one, since it shouldn't have any reliance on `@celo/phone-number-privacy-common`. That said, this isn't that much different than the point of `@celo/base` anyways, which is why I opted for this first, though happy to split it up instead!

Happy to explore other options if you have other ideas @aaronmgdr @alecps !

### Changes
- remove `getPhoneHash` in `@celo/base` in favor of `getIdentifierHash`; have implementations in `@celo/phone-utils` & `@celo/identity` depend on base; also moves prefixes to base in order to do this
  - remove `attestations`-specific code in `@celo/base`
  - add simple unit tests for `getIdentifierHash` in base package that should force future PRs that add a new prefix to also add the expected unit test value
- modifies `generate-old-devchain-and-build.sh` to build the entire monorepo when generating the old released devchain to get `pre-protocol-test-release` & `protocol-test-release` tests to pass. Note:
  - CK sneakily sets `BUILD_DIR` to use the `RELEASE_TAG` when this is set, which caused sneaky build failures in CI when this was built
  - it's necessary to build `packages/protocol` entirely (typescript & solidity); if we only build typechain types + solidity, the migrations are not transpiled and none are run when creating the devchain, which causes the pre-test to succeed but the later `protocol-test-release` to fail.

NOTE: this PR doesn't remove all ASv1-related code from the SDK (for readability here, where there are more structural SDK changes).

### Tested
- unit tests
- CI

### CODE REVIEW

File: packages/phone-number-privacy/combiner/test/integration/legacypnp.test.ts [hunk 1]
```diff
@@ -12,6 +11,7 @@ import {
   TestUtils,
   WarningMessage,
 } from '@celo/phone-number-privacy-common'
+import { AttestationsStatus } from '@celo/phone-number-privacy-common/lib/test/utils'
```
Comment (aaronmgdr) [inline packages/phone-number-privacy/combiner/test/integration/legacypnp.test.ts:14]: just checking that this is supposed importing from test?

it doesnt seem test related

Comment (eelanagaraj) [reply packages/phone-number-privacy/combiner/test/integration/legacypnp.test.ts:14]: yup! (these are normally used from `TestUtils` but was running into type errors trying to import the interface that way)

this file + all of the `AttestationsStatus` stuff should be deleted after the `LEGACY_PNP` endpoint is fully deprecated :)

File: packages/sdk/base/src/identifier.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export const PEPPER_SEPARATOR = '__'
+
+/**
+ * Standardized prefixes for ODIS identifiers.
+ * Used in @celo/identity, @celo/phone-utils, and the protocol package.
+ *
+ * When adding prefixes, make sure to add the expected value for the unit test case
+ * in identifier.test.ts, otherwise the test will fail.
+ */
+export enum IdentifierPrefix {
```
Comment (alecps) [inline packages/sdk/base/src/identifier.ts:31]: why did we move this to the base package?

**Review by alecps — COMMENTED** [review]
I trust that you've thought through this, but at first glance it seems like a move in the wrong direction to be adding the identifier code and prefixes into the base package. Could you add an explanation of why you're doing this to the PR description?

**Comment by eelanagaraj** [comment]
@alecps @aaronmgdr added some more detailed description, lmk your thoughts!

File: packages/sdk/base/src/identifier.ts (hunk 2 above)
Comment (eelanagaraj) [reply packages/sdk/base/src/identifier.ts:31]: I did this to maintain the structure of these functions & minimize duplication; `getIdentifierHash` -> `getPrefixedIdentifier` -> `IdentifierPrefix`, and `phone-utils` also needs the `IdentifierPrefix.PHONE_NUMBER`. I laid out a couple other alternatives that are possible in the PR description now though, lmk what you think + happy to discuss in more detail!

**Review by alecps — APPROVED** [review]
Nice!!

**Pull request merged**